src/modules/bulk-minter/components/change/ChangeDonationField.tsx

```diff
@@ -0,0 +1,280 @@
+      <StyledCreatorsRow className="minter-charity-row">
+        {icon_url ? (
+          <img height={32} width={32} className="rounded-full" src={icon_url} alt={name + '-logo'} />
+        ) : (
+          <img height={32} width={32} className="rounded-full"  src={creatorStandinImg} alt="generic-logo" />
+        )} 
+        <Paragraph
+          style={{
+            margin: '0 14px 0 6px',
+            maxWidth: 200,
+            fontSize: 14,
+            lineHeight: 1.2,
+          }}
+        >
+          {charity.name}
+        </Paragraph>
+        <FeatherIcon
+            className="creator-row-icon"
+            icon="external-link"
+            onClick={() => {window.open(CHANGE_BASE_URL + solana_address, "_blank")}}
+          />
+        <span style={{ marginLeft: 'auto' }}></span>
+        { enableSelectButton && <StyledSelectButton
+          onClick={onSelect}
+        >
+          {selectButtonText}
+        </StyledSelectButton> }
+        { !enableSelectButton && <AlreadyAddedNotice
+        >
+          {selectButtonText}
+        </AlreadyAddedNotice> }
+      </StyledCreatorsRow>  
+        <Paragraph
+          style={{ fontSize: 10, marginLeft:14, lineClamp: 4, textOverflow: 'ellipsis', overflow: 'hidden' }}
+        >
+          {description}
+        </Paragraph>
+        <Paragraph
+          style={{ fontSize: 10, marginLeft:14 }}
+          >
+          EIN: {ein}
+        </Paragraph>
+        </div>
+    );
+  };
+
+
+export const ChangeDonationField: FunctionComponent<Props> = ({
+    showDonationField,
+    creators,
+    updateDonationFieldState,
+    updateCreatorState,
+    updateShowErrorState
+  }: Props) => {
+    const [searchResults, setSearchResults] = useState<ChangeNonprofit[]>([]);
+    const [charityLoading, setLoading] = useState<boolean>(false);
+    const [isTyping, setIsTyping] = useState<boolean>(false);
+    const debounce = useMemo(() => new Debounce(), []);
+  
+    const addDonation = (nonProfit: ChangeNonprofit) => {
+        if (creators.length >= MAX_CREATOR_LIMIT) {
+          throw new Error('Max level of creators reached');
+        }
+        const existingCreators = creators.map((c) => c.address);
+        const indexOfDuplicate = existingCreators.findIndex((a) => nonProfit.crypto.solana_address === a);
+        if (indexOfDuplicate !== -1) {
+          throw new Error('All creator hashes must be unique');
+        }
+        const newShareSplit = 98 / (creators.length + 1);
+        updateCreatorState([
+          ...creators.map((c) => ({ ...c, share: newShareSplit })),
+          { address: nonProfit.crypto.solana_address, 
+            share: newShareSplit,
+            charityProps: {
+              isCharity: true,
+              displayName: nonProfit.name,
+              imageUrl:nonProfit.icon_url
+            } },
+        ]);
+        setSearchResults([]);
+        updateDonationFieldState(false);
+        updateShowErrorState(false);
+      };
```

Comment:
If using `useCallback`

```suggestion
      }, [setSearchResults]); // may need to add other dependencies
```